### PR TITLE
Deprecate snow/ice cover

### PR DESCRIPTION
**Pull request by m-mohr**
As it's available in eo as eo:snow_cover

### CODE REVIEW

**Review by gadomski — APPROVED** [review]

File: json-schema/schema.json [hunk 1]
```diff
@@ -61,7 +61,6 @@
         { "required": ["s2:generation_time"] },
         { "required": ["s2:processing_baseline"] },
         { "required": ["s2:water_percentage"] },
-        { "required": ["s2:snow_ice_percentage"] },
```
Comment (philvarner) [inline json-schema/schema.json:64]: elsewhere it says deprecated, but it was removed here.

File: examples/item2.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
       "msi"
     ],
     "eo:cloud_cover": 51.580326,
+    "eo:snow_cover": 0,
```
Comment (philvarner) [inline examples/item2.json:24]: one thing to consider is if we want to have Yet Another Extension (if we don't already have one, there are so many!) for scene analytics, and we move the eo cloud_cover and snow_cover fields there, along with no data percentage and other descriptions of the scene. But for this change, this is the most reasonable approach right now.

File: examples/item2.json [hunk 3]
```diff
@@ -55,7 +56,6 @@
     "s2:medium_proba_clouds_percentage": 14.61311,
```
[the remark this answers is not in the document]
Comment (philvarner) [reply examples/item2.json:56]: the comment doesn't show what I selected well -- everything from s2:reflectance_conversion_factor to "s2:thin_cirrus_percentage"

File: json-schema/schema.json (hunk 1 above)
Comment (m-mohr) [reply json-schema/schema.json:64]: I felt like people should provide at least one field that's not deprecated.

File: examples/item2.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (m-mohr) [reply examples/item2.json:56]: Yeah, I think we habe an issue somewhere to discuss this. Can't quite recall where this was though.

**Pull request merged**